llvm/lib/Target/DirectX/DXILOpLowering.cpp

```diff
@@ -770,8 +770,14 @@ class OpLowerer {
         continue;
       Intrinsic::ID ID = F.getIntrinsicID();
       switch (ID) {
-      default:
+      case Intrinsic::dx_resource_casthandle:
```

Comment:
Might be worth commenting that casthandle is okay because we handle it specifically at the end.